components/Addressing/models.rst

```diff
@@ -1,262 +1,222 @@
++-----------+-----------------------------------------+
 
-    foreach ($zone->getMembers() as $member) {
-        echo $member->getName();
-    }
+.. note::
+   This model implements :ref:`component_addressing_model_zone-member-interface` |br|
+   For more detailed information go to `Sylius API ZoneMember`_.
 
-    $zone->getCreatedAt();
-    $zone->getUpdatedAt();
+.. _Sylius API ZoneMember: http://api.sylius.org/Sylius/Component/Addressing/Model/ZoneMember.html
 
-Three different types of zones are supported out-of-the-box.
+.. note::
+   Each **ZoneMember** instance holds a reference to the **Zone** object and
+   an appropriate area entity, for example a **Country**.
 
-* ``country`` zone, which consists of many countries.
-* ``province`` zone, which is constructed from multiple provinces.
-* ``zone``, which is a group of other zones.
+There are three default zone member models:
 
-Each zone type has different **ZoneMember** model, but they all expose the same API:
+* :ref:`component_addressing_model_zone-member-country`
+* :ref:`component_addressing_model_zone-member-province`
+* :ref:`component_addressing_model_zone-member-zone`
 
-.. code-block:: php
-
-    <?php
+.. tip::
+   Feel free to implement your own custom zone members!
 
-    foreach ($zone->getMembers() as $member) {
-        echo $member->getName();
+.. _component_addressing_model_zone-member-country:
 
-        echo $member->getZone()->getName(); // Name of the zone.
-    }
+ZoneMemberCountry
+-----------------
 
-There are following models and each of them represents a different zone member:
+Country member of a zone is represented as a **ZoneMemberCountry** instance.
+It has all the properties of :ref:`component_addressing_model_zone-member` and one additional:
 
-* ``ZoneMemberCountry``
-* ``ZoneMemberProvince``
-* ``ZoneMemberZone``
++----------+--------------------------------------------------+
+| Property | Description                                      |
++==========+==================================================+
+| country  | The **Country** associated with this zone member |
++----------+--------------------------------------------------+
 
-Each **ZoneMember** instance holds a reference to the **Zone** object and appropriate area entity, for example a **Country**.
+.. note::
+   For more detailed information go to `Sylius API ZoneMemberCountry`_.
 
-Creating a zone requires adding appropriate members:
+.. _Sylius API ZoneMemberCountry: http://api.sylius.org/Sylius/Component/Addressing/Model/ZoneMemberCountry.html
 
-.. code-block:: php
+.. _component_addressing_model_zone-member-province:
 
-    <?php
+ZoneMemberProvince
+------------------
 
-    use Sylius\Component\Addressing\Country;
-    use Sylius\Component\Addressing\Zone;
-    use Sylius\Component\Addressing\ZoneInterface;
-    use Sylius\Component\Addressing\ZoneMemberCountry;
+Province member of a zone is represented as a **ZoneMemberProvince** instance.
```

Comment:
Just checked, everywhere in **ZoneMember** area, everything above is alright. My best guess is, there was another word which i changed and forgot about it ._.